realm/realm-library/src/main/cpp/jni_util/java_global_ref_copy.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef REALM_JNI_UTIL_JAVA_GLOBAL_REF_COPY_HPP
+#define REALM_JNI_UTIL_JAVA_GLOBAL_REF_COPY_HPP
+
+#include <jni.h>
+
+namespace realm {
+    namespace jni_util {
+
+        // Manage the lifecycle of jobject's global ref via copy constructors
```

Comment:
I would always add some justification as to why this could be used (e.g. the description you have in the PR).